### ISSUE TITLE
Semi-cryptic error when parsing boolean column with non-boolean values

### ISSUE THREAD

**Issue by phargogh** [natcap/invest#1449]
A [user on the forums posted about encountering this issue](https://community.naturalcapitalproject.org/t/error-in-running-urban-cooling-model/4218).  In the Urban Cooling Model, they were trying to provide values of `0.3` to the boolean `green_space` column, and the resulting exception was:

```python
Traceback (most recent call last):
  File "natcap\invest\utils.py", line 164, in prepare_workspace
  File "invest\cli.py", line 470, in main
  File "natcap\invest\urban_cooling_model.py", line 415, in execute
  File "natcap\invest\utils.py", line 701, in read_csv_to_dataframe
  File "pandas\core\generic.py", line 6532, in astype
  File "pandas\core\internals\managers.py", line 414, in astype
  File "pandas\core\internals\managers.py", line 354, in apply
  File "pandas\core\internals\blocks.py", line 616, in astype
  File "pandas\core\dtypes\astype.py", line 238, in astype_array_safe
  File "pandas\core\dtypes\astype.py", line 183, in astype_array
  File "pandas\core\dtypes\astype.py", line 80, in _astype_nansafe
  File "pandas\core\arrays\masked.py", line 149, in _from_sequence
  File "pandas\core\arrays\boolean.py", line 355, in _coerce_to_array
  File "pandas\core\arrays\boolean.py", line 190, in coerce_to_array
TypeError: Need to pass bool-like values
```

The text of the exception is enough to point someone in kind of the right direction, but it would be great if maybe we could catch exceptions like this that have to do with column type mismatches and include some extra information to help with debugging.

**Comment by emlys** [comment]
@phargogh I think I can fix this in #1379. There's supposed to be a more verbose message: `'Value(s) in the "{col}" column of the table {csv_path} could not be interpreted as {col_spec["type"]}s. Original error: {err}'`. But this is only catching `ValueError`s, and the boolean issue is a `TypeError`. I'll update it to catch any type of exception.

**Comment by phargogh** [comment]
Thanks @emlys !  I'll mark this issue as a duplicate.
